org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/util/JavaDocHelper.xtend

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2018 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xbase.ui.util
+
+import com.google.inject.Inject
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.jface.text.IRegion
+import org.eclipse.jface.text.Region
+import org.eclipse.xtext.common.types.TypesPackage
+import org.eclipse.xtext.documentation.IJavaDocTypeReferenceProvider
+import org.eclipse.xtext.naming.IQualifiedNameConverter
+import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.scoping.IScopeProvider
+import org.eclipse.xtext.util.Pair
+import org.eclipse.xtext.util.Tuples
+
+import static extension org.eclipse.xtext.nodemodel.util.NodeModelUtils.findActualSemanticObjectFor
+import static extension org.eclipse.xtext.nodemodel.util.NodeModelUtils.findLeafNodeAtOffset
+
+/**
+ * @author miklossy - Initial contribution and API
+ * 
+ * @since 2.16
+ */
+class JavaDocHelper {
+
+	@Inject extension IScopeProvider
+
+	@Inject extension IJavaDocTypeReferenceProvider
```

Comment:
Judging from the names of the types, the method `getReferencedElementInJavaDoc` belongs to the IJavaDocTypeReferenceProvider - which would deserve a better name, too since it does not know anything about types at a first glance.

@miklossy @cdietrich 
What do you think about a new default method on `IJavaDocTypeReferenceProvider` that returns something like `class EObjectInComment { EObject object; ITextRegion region }` and returns nothing in its default impl (in the default method of the interface). The MultiLineComment thingy would have a template method `getEReference():EReference` that returns null and has the bigger part of the logic as it is implemented in this new helper right now. We could have a special binding for Xbase that is aware of `TypesPackage.Literals.JVM_PARAMETERIZED_TYPE_REFERENCE__TYPE` and than we would not need a new helper class and spread the handling of comments accross different abstractions.